tests/plotting/test_plotting.py

```diff
@@ -1647,7 +1647,8 @@ def test_opacity_by_array_uncertainty(uniform):
 
 
 def test_opacity_by_array_user_transform(uniform):
-    uniform['Spatial Point Data'] /= uniform['Spatial Point Data'].max()
+    scaled = uniform['Spatial Point Data'] / uniform['Spatial Point Data'].max()
+    uniform['Spatial Point Data'] = scaled
```

Comment:
Why was this change necessary?

Reply:
This was the most perplexing part of this PR, and I was going to gloss over it until I figured it out, but you're to good for anything to sneak past you.

Accessing an array directly from a dataset actually returns a `numpy.ndarray` rather than a `pyvista.pyvista_ndarray` due to the way that `pyvista.utilities.helpers.point_array` works. For whatever reason, operating on a `numpy.ndarray` representation of our data leaves an uncollected VTK array.

This has been corrected in ea76919e6b7bacc42b659ae86c6de8bc74f42bbd, but no we have to deal with the problem that `pyvista_ndarray.max` returns a one element array that is treated differently than a size 1 `numpy.ndarray`. This was already discussed in #1953 and commented [here](https://github.com/pyvista/pyvista/pull/1953#issuecomment-1146896567).

This will have to be resolved in a separate PR first.

Reply:
Flattery will get you nowhere :laughing: For what it's worth I would've been OK with keeping this as-is and postponing its investigation for later.

So if part of the issue was that
> Accessing an array directly from a dataset actually returns a `numpy.ndarray` rather than a `pyvista.pyvista_ndarray` due to the way that `pyvista.utilities.helpers.point_array` works
do we want to change that? Or is that behaviour necessary/expected/etc.?

Reply:
I’ve changed it in #2770. We should always return pyvista_ndarray.